qos-service-impl/src/main/java/com/palantir/atlasdb/qos/ratelimit/CassandraMetricsClientLimitMultiplier.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.qos.ratelimit;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import com.palantir.atlasdb.qos.config.CassandraHealthMetricMeasurement;
+import com.palantir.atlasdb.qos.config.QosCassandraMetricsInstallConfig;
+import com.palantir.atlasdb.qos.config.QosCassandraMetricsRuntimeConfig;
+import com.palantir.atlasdb.qos.config.QosPriority;
+import com.palantir.atlasdb.qos.config.ThrottlingStrategy;
+import com.palantir.cassandra.sidecar.metrics.CassandraMetricsService;
+import com.palantir.remoting3.clients.ClientConfigurations;
+import com.palantir.remoting3.jaxrs.JaxRsClient;
+
+public class CassandraMetricsClientLimitMultiplier implements ClientLimitMultiplier {
+    private ThrottlingStrategy throttlingStrategy;
+    private CassandraMetricMeasurementsLoader cassandraHealthMetrics;
+
+    private CassandraMetricsClientLimitMultiplier(
+            ThrottlingStrategy throttlingStrategy,
+            CassandraMetricMeasurementsLoader cassandraHealthMetrics) {
+        this.throttlingStrategy = throttlingStrategy;
+        this.cassandraHealthMetrics = cassandraHealthMetrics;
+    }
+
+    public static ClientLimitMultiplier create(Supplier<QosCassandraMetricsRuntimeConfig> runtimeConfig,
+            QosCassandraMetricsInstallConfig installConfig) {
+        CassandraMetricsService metricsService = JaxRsClient.create(
+                CassandraMetricsService.class,
+                "qos-service",
+                ClientConfigurations.of(installConfig.cassandraServiceConfig()));
```

Comment:
do we need metrics instrumentation wrapping calls to qos service?

Reply:
These are calls to the Cassandra-sidecar service, but yes I think it would be good to have metrics on the numRequests, failures etc. Can you point me to the instrumentation wrapper here?